tutoraspects/templates/openedx-assets/queries/at_risk_learner_filter.sql

```diff
@@ -0,0 +1,18 @@
+with
+    page_visits as (
+        select org, course_key, actor_id, max(last_visited)
+        from {{ ASPECTS_XAPI_DATABASE }}.fact_learner_last_course_visit
+        where
+            1=1
+            {% include 'openedx-assets/queries/common_filters.sql' %}
+            and last_visited < subtractDays(now(), 7)
+        group by org, course_key, actor_id
+    )
```

Comment:
Data is already grouped/merged on the downstream MV. Also, as the table is partitioned per org/course_key you don't need to add the common filters.

Reply:
The grouping isn't entirely true, we still need it to narrow down unmerged blocks: 

> Data deduplication occurs only during a merge. Merging occurs in the background at an unknown time, so you can’t plan for it. Some of the data may remain unprocessed. Although you can run an unscheduled merge using the OPTIMIZE query, do not count on using it, because the OPTIMIZE query will read and write a large amount of data.
> 
> Thus, ReplacingMergeTree is suitable for clearing out duplicate data in the background in order to save space, but it does not guarantee the absence of duplicates.

The common filters is what makes this more performant (at least until the new CH analyzer lands in CH Cloud), otherwise it will pull all rows here and filter them in the join step below.

Reply:
`last_visited` needs to be renamed to `emission_time` in the select statement